### ISSUE TITLE
Test pip install in CI

### ISSUE THREAD

**Issue by jtniehof** [spacepy/spacepy#542]
Our CI focuses on making sure the code works. This doesn't help much when pip or setuptools changes around and breaks the installer itself (#530). The CI should do a pip install from the latest git, e.g. `pip install git+https://github.com/spacepy/spacepy`, but make sure it has the special sauce to go from the merge commit for a PR, rather than the default branch.

We probably need to also run all the unit tests after the install as well, since the installer only rarely fails in the sense of returning non-zero exit status.

**Comment by jtniehof** [comment]
Note we will need to add explicit setuptools versioning to the test environments as well, just like over dependencies.

**Comment by dstansby** [comment]
I agree my PR wasn't ideal 😄  One option would be to use [`tox`](https://tox.wiki/en/latest/index.html), which you can use for (quoting from their page):

- checking that your package installs correctly with different Python versions and interpreters
- running your tests in each of the environments, configuring your test tool of choice
- acting as a frontend to Continuous Integration servers, greatly reducing boilerplate and merging CI and shell-based testing.

We use `tox` over in `sunpy` and I've used it in other projects, and would definitely recommend it!
